packages/app-admin-ui/client/components/CreateItemModal.js

```diff
@@ -48,12 +48,14 @@ class CreateItemModal extends Component {
       return;
     }
 
-    const nonPrimaryKeyFields = fields.filter(({ isPrimaryKey }) => !isPrimaryKey);
+    const creatable = fields
+      .filter(({ isPrimaryKey }) => !isPrimaryKey)
```

Comment:
~Is `isPrimaryField` an optional `boolean` that is always `true` if it exists on an object?
If so then this part of the function tells me that we are creating an array of `field`s who's~
This is hard for me to follow, what is `createTable` doing?

Reply:
I'm going to guess that the first filter returns items without a `isPrimaryKey` prop, which when exists is always true.

Reply:
Yep, that's right; it checks to see if `isPrimaryKey` exists and/or is not truthy. (ie; we want to filter out anything which might be flagged as `isPrimaryKey`)